tests/js/scroll/scroll-bidirectional-tests.js

```diff
@@ -0,0 +1,249 @@
+/*
+Copyright (c) 2020 The Gamepad Navigator Authors
+See the AUTHORS.md file at the top-level directory of this distribution and at
+https://github.com/fluid-lab/gamepad-navigator/raw/master/AUTHORS.md.
+
+Licensed under the BSD 3-Clause License. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the BSD 3-Clause License at
+https://github.com/fluid-lab/gamepad-navigator/blob/master/LICENSE
+*/
+
+/* global gamepad, jqUnit */
+
+(function (fluid, $) {
+    "use strict";
+
+    $(document).ready(function () {
+
+        fluid.registerNamespace("gamepad");
+        fluid.registerNamespace("gamepad.tests");
+
+        // Define the required variables.
+        gamepad.tests.windowObject = window;
+        gamepad.tests.frequency = 50;
+
+        jqUnit.module("Gamepad Navigator Bidirectional Scrolling Tests");
+
+        jqUnit.asyncTest("Scroll horizontally", function () {
+            jqUnit.expect(5);
+
+            // No of times the gamepad mock should be returned.
+            gamepad.tests.windowObject.count = 10;
+            gamepad.tests.windowObject.navigator.getGamepads = function () {
+                return gamepad.tests.utils.bidirectional.oneAxis(0);
+            };
+            $(window).scrollLeft(0);
+            jqUnit.assertEquals("The vertical scroll position should be at the left of the webpage.", 0, $(window).scrollLeft());
+
+            var testNavigator = gamepad.inputMapper({
+                windowObject: gamepad.tests.windowObject,
+                frequency: gamepad.tests.frequency
+            });
+            jqUnit.assertValue("The Gamepad Navigator should be instantiated.", testNavigator);
+
+            // Trigger gamepadconnected event and start scrolling horizontally.
+            testNavigator.events.onGamepadConnected.fire();
+            var xPosition = null;
+            gamepad.tests.intervalLoop = setInterval(function () {
+                if (gamepad.tests.windowObject.count === 4) {
+                    xPosition = $(window).scrollLeft();
+                    jqUnit.assertNotEquals("The page should have scrolled right.", 0, xPosition);
+                }
+                else if (gamepad.tests.windowObject.count === 0) {
+                    var isScrolledLeft = $(window).scrollLeft() < xPosition ? true : false;
+                    jqUnit.assertTrue("The page should have scrolled left.", isScrolledLeft);
+
+                    testNavigator.events.onGamepadDisconnected.fire();
+                    clearInterval(gamepad.tests.intervalLoop);
+                    testNavigator.destroy();
+                    jqUnit.assertTrue("The instance of the gamepad navigator should be destroyed.", fluid.isDestroyed(testNavigator));
+                    jqUnit.start();
+                }
+            }, gamepad.tests.frequency);
```

Comment:
Pitting one interval against another seems like it could easily lead to race conditions once you start trying to test more complex sequences of keys, but as long as the test frequency is significantly longer than the gamepad polling frequency I guess it's fine for now.